app/controllers/staff/reports_state_controller.rb

```diff
@@ -24,89 +32,40 @@ def edit
   def update
     case report.state
     when "inactive"
-      change_report_state_to_active
+      change_report_state_to(:active)
     when "active"
-      change_report_state_to_submitted
+      change_report_state_to(:submitted)
     when "submitted"
-      change_report_state_to_in_review
+      change_report_state_to(:in_review)
     when "in_review"
-      params[:request_changes] ? change_report_state_to_awaiting_changes : change_report_state_to_approved
+      params[:request_changes] ? change_report_state_to(:awaiting_changes) : change_report_state_to(:approved)
     when "awaiting_changes"
-      change_report_state_to_submitted
+      change_report_state_to(:submitted)
     else
       authorize report
       redirect_to report_path(report)
     end
   end
 
-  private def confirm_activation
-    authorize report, :activate?
-    @report_presenter = ReportPresenter.new(report)
-    render "staff/reports_state/activate/confirm"
-  end
-
-  private def change_report_state_to_active
-    authorize report, :activate?
-    report.update!(state: :active)
-    report.create_activity key: "report.activated", owner: current_user
-    @report_presenter = ReportPresenter.new(report)
-    render "staff/reports_state/activate/complete"
-  end
-
-  private def confirm_submission
-    authorize report, :submit?
-    @report_presenter = ReportPresenter.new(report)
-    render "staff/reports_state/submit/confirm"
-  end
-
-  private def change_report_state_to_submitted
-    authorize report, :submit?
-    report.update!(state: :submitted)
-    report.create_activity key: "report.submitted", owner: current_user
-    @report_presenter = ReportPresenter.new(report)
-    render "staff/reports_state/submit/complete"
-  end
-
-  private def confirm_in_review
-    authorize report, :review?
-    @report_presenter = ReportPresenter.new(report)
-    render "staff/reports_state/review/confirm"
-  end
-
-  private def change_report_state_to_in_review
-    authorize report, :review?
-    report.update!(state: :in_review)
-    report.create_activity key: "report.in_review", owner: current_user
-    @report_presenter = ReportPresenter.new(report)
-    render "staff/reports_state/review/complete"
-  end
-
-  private def confirm_request_changes
-    authorize report, :request_changes?
-    @report_presenter = ReportPresenter.new(report)
-    render "staff/reports_state/request_changes/confirm"
-  end
-
-  private def change_report_state_to_awaiting_changes
-    authorize report, :request_changes?
-    report.update!(state: :awaiting_changes)
-    report.create_activity key: "report.awaiting_changes", owner: current_user
+  private def confirm_state(state)
+    authorize report, state + "?"
     @report_presenter = ReportPresenter.new(report)
-    render "staff/reports_state/request_changes/complete"
+    render "staff/reports_state/#{state}/confirm"
```

Comment:
There's a similar problem here with assuming the state name is the same as the names of the views, which I'm not sure is true. This has a lot of overlap with the changes I've made in https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/pull/616/commits/ae79d1a2519305075aaccaaa8859a9c62992f56c, which extracts policy names, log actions, and view names into constants so they can be looked up by state name.

Reply:
As you have taken this further why don't I drop the refactor and you add your work from #616 ontop of this?

Reply:
Ah sorry, this was my mistake in reading the code: the parameter to this method is named `state`, but it's not a state name, it's an action, e.g. `review` rather than `in_review`. That means the `authorize` and `render` calls are correct but the code could be clarified, either by renaming the `state` parameter here, or making `confirm_state` and `change_report_state_to` _both_ take state names.

Reply:
Yeah, I was going to say that 'confirm' methods needs better langauge, but I think dropping my refactor and applying yours is the better option? Thoughts?

Reply:
I think it would make sense to back this refactor out of this PR, and combine it with some of my changes in #616. I like what you've done here, would just benefit from a bit of renaming and being reviewed separately from all the functional changes in this PR.

Reply:
(It's actually less complicated than what I attempted in #616, it doesn't require quite as much config/constants mapping states to other things as I've done.)

Reply:
Okay cool, I only glanced at #616, but I am interested in the race condition (plus I just love to try and use that phrase!) I've dropped the refactor on here. I think without the refactor this still stands and could go in as long as we follow up by making sure we handle your concerns.

Reply:
Absolutely :) Do open your refactoring as a new PR and we'll land that before I have another look at #616. I'm sorry, I should have checked in with you before I started my PR, I'd assumed that controller was "finished" and wouldn't be being changed much.

Reply:
No probs at all! 👍 I'll open a PR but I'm interested in why you want to pass the state as a parm so it may be #616 makes more sense!

Reply:
Although I had the idea for this refactoring while turning `state` into a form parameter, it really doesn't have to be done in my PR, it can be done separately. Let's get your refactoring of the `confirm` and `change` methods in and then I'll rebase #616.